app/controllers/authenticate_controller.rb

```diff
@@ -40,25 +33,54 @@ def authenticate
     )
     render json: authentication_token
   rescue => e
-    logger.debug("Authentication Error: #{e.message}")
-    e.backtrace.each do |line|
-      logger.debug(line)
-    end
-    raise Unauthorized
+    handle_authentication_error(e)
   end
 
+  def authenticate_oidc
+    authentication_token = new_authentication_strategy.conjur_token_oidc(
+      ::Authentication::Strategy::Input.new(
+        authenticator_name: 'authn-oidc',
+        service_id:         params[:service_id],
+        account:            params[:account],
+        username:           nil,
+        password:           nil, #TODO: the body will contain info about OpenID
+        origin:             request.ip,
+        request:            request
+      )
+    )
+    render json: authentication_token
+  rescue => e
+    handle_authentication_error(e)
+  end
+
+
   def k8s_inject_client_cert
     ::Authentication::AuthnK8s::Authenticator.new(env: ENV).inject_client_cert(params, request)
     head :ok
   rescue => e
+    handle_authentication_error(e)
+  end
+
+  private
+
+  def handle_authentication_error(e)
     logger.debug("Authentication Error: #{e.message}")
     e.backtrace.each do |line|
       logger.debug(line)
     end
     raise Unauthorized
   end
 
-  private
+  def new_authentication_strategy
```

Comment:
Pulling this out was a good move, and it can even memoized with the standard memoization pattern:
```
@new_authentication_strategy ||= ::Authentication::Strategy... 
```
It could even be created in a rails initializer though that's something I'd like to think through a bit before doing...

Reply:
done